internal/runner/runner.go

```diff
@@ -200,15 +200,31 @@ func (r *Runner) Start() error {
 		return err
 	}
 
-	// Start the watcher and the gorotuine that receives configs
-	ch := make(chan *pb.RunnerConfig)
-	go r.watchConfig(r.runningCtx, ch)
-	go r.recvConfig(r.runningCtx, client, ch)
-
+	errCh := make(chan error)
+
+	// Start the watcher and the goroutine that receives configs
+	configCh := make(chan *pb.RunnerConfig)
+	go func() {
+		r.watchConfig(r.runningCtx, configCh)
+		errCh <- status.Errorf(codes.Internal, "Stopped watching for config changes")
```

Comment:
I would add an errCh to the argument list of `watchConfig` and `recvConfig` instead of this.